substrate/client/src/client.rs

```diff
@@ -213,9 +214,12 @@ impl<B, E, Block> Client<B, E, Block> where
 
 	/// Get the set of authorities at a given block.
 	pub fn authorities_at(&self, id: &BlockId<Block>) -> error::Result<Vec<AuthorityId>> {
-		self.executor.call(id, "authorities", &[])
-			.and_then(|r| Vec::<AuthorityId>::decode(&mut &r.return_data[..])
-				.ok_or(error::ErrorKind::AuthLenInvalid.into()))
+		match self.backend.blockchain().cache().and_then(|cache| cache.authorities_at(*id)) {
```

Comment:
Isn't `cache()` returning [a `None` on regular clients](https://github.com/paritytech/polkadot/pull/251/files#diff-ca3049c7e7cc28b90c4b65bf956f7c1cR210) and thus not ever triggering the `.and_then` and consequentially not calling the executor anymore, breaking old behavior?

Reply:
Executor is called when `cache().and_then()` resolves to `None` - it is the case for full clients or when the entry is missing on light clients